### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by sirkon**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,74 @@
...
+## Design
+
+This tool consists of two parts:
+
+1. Cache generator `apriori generate`
+2. Go modules proxy `apriori serve`
+
+### How cache generator works
+
+Cache generator recieves a list of modules to store in cache. The list should looks like
```
Comment (vitalyisaev2) [inline README.md]: should look

**Review by zeslava — APPROVED** [review]

File: args-processor.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+			if _, err := dest.Write(data); err != nil {
+				ap.logger.Fatal().Err(err).Msg("failed to save apriori info")
+			}
+			ap.logger.Info().Msg("apriori generation done")
+		}
+	}
+}
+
+func (ap *argsProcessor) getModule(module string) (goproxy.Module, error) {
+	req, _ := http.NewRequest("GET", fmt.Sprintf("http://society.org/%s/@v/list", module), nil)
```
Comment (vitalyisaev2) [inline args-processor.go]: society.org? O_O

**Review by vitalyisaev2 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (sirkon) [reply README.md]: fixed

File: args-processor.go (hunk 2 above)
Comment (sirkon) [reply args-processor.go]: It does nothing. We just need `http.Request` to get `goproxy.Module` from `gorpoxy.Plugin`

**Review by vitalyisaev2 — APPROVED** [review]

**Review by alyoshka — APPROVED** [review]

**Pull request merged**